### PR TITLE
User Profiles

### LINKED ISSUE #677: Finish profiles prototype
PR is here: #679  ~#573~
- [x] rebase to staging
- [x] add / check comments
- [x] activities are not done api wise, so ignore for now
- [x] routing: check `/user/profile/{username}` (some info here: #576) 
- [x] routing: decide where `/user/me` and `/user/public` are handled
- [x] create issue for CFWorker to allow type user to be displayed in new FE ([issue](https://github.com/serlo/serlo.org-cloudflare-worker/issues/65))

### PR DESCRIPTION
**Pull request by elbotho**
rebase of #573

### example profile
https://frontend-git-profiles-rebase.serlo.vercel.app/user/profile/wolfgang
https://frontend-git-profiles-rebase.serlo.vercel.app/user/profile/kathongi

### issue
closes https://github.com/serlo/frontend/issues/677

### status
- Uses mockup data for now (because descriptions are not part of the staging data (returns NULL)

### CODE REVIEW

**Comment by inyono** [comment]
FYI: staging has anonymized data (e.g. removes all email addresses & descriptions of users). This would work in prod.

**Comment by elbotho** [comment]
@Entkenntnis okay! This is ready for review now. 
As per your suggestion I implemented the link to /user/profile/username in the menu (best solution) and added a simple fallback redirect for /user/me and /user/profile that we can remove after 100%.

File: src/components/content/author-tools.tsx [hunk 1]
```diff
@@ -30,6 +30,7 @@ export function AuthorTools({ data }: AuthorToolsProps) {
       content={<AuthorToolsHoverMenu data={data} />}
     >
       <EditButton>
+        author tools
```
Comment (Entkenntnis) [inline src/components/content/author-tools.tsx]: Hm, looks superfluous here

File: src/components/pages/user/profile-community-banner.tsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+      </ExplanationsWrapper>
+    )
+  }
+
+  function renderExplanationText(
+    key: 'activeDonors' | 'activeAuthors' | 'activeReviewers'
+  ) {
+    if (isOwnProfile) {
+      const counts = fetchGroupsTotalCounts()
+      const parts = bannerContent[key].ownProfileMessage.split('%no%')
```
Comment (Entkenntnis) [inline src/components/pages/user/profile-community-banner.tsx]: We have something better now :)

File: src/components/pages/user/profile-community-banner.tsx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+          <b>{counts && counts[key].totalCount}</b>
+          {parts[1]}
+        </>
+      )
+      return <StyledP>{ownProfileMessage}</StyledP>
+    }
+
+    return (
+      <StyledP>
+        {bannerContent[key].otherUserProfileMessage.replace(
```
Comment (Entkenntnis) [inline src/components/pages/user/profile-community-banner.tsx]: Same here :)

**Review by Entkenntnis — CHANGES_REQUESTED** [review]
Looks fine! (Finally getting this done)

**Review by Entkenntnis — APPROVED** [review]

**Comment by elbotho** [comment]
Thanks @Entkenntnis

**Pull request merged**